sdl_android/src/main/java/com/smartdevicelink/api/FileManager.java

```diff
@@ -0,0 +1,395 @@
+			int fileNum = 0;
+
+			@Override
+			public void addCorrelationId(int correlationid) {
+				super.addCorrelationId(correlationid);
+				if(deletionOperation){
+					fileNameMap.put(correlationid, ((DeleteFile) requests.get(fileNum++)).getSdlFileName());
+				}else{
+					fileNameMap.put(correlationid, ((PutFile) requests.get(fileNum++)).getSdlFileName());
+				}
+			}
+
+			@Override
+			public void onUpdate(int remainingRequests) {}
+
+			@Override
+			public void onFinished() {
+				if(errors.isEmpty()){
+					listener.onComplete(null);
+				}else{
+					listener.onComplete(errors);
+				}
+			}
+
+			@Override
+			public void onError(int correlationId, Result resultCode, String info) {
+				if(fileNameMap.get(correlationId) != null){
+					errors.put(fileNameMap.get(correlationId), buildErrorString(resultCode, info));
+				}// else no fileName for given correlation ID
+			}
+
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if(response.getSuccess()){
+					if(fileNameMap.get(correlationId) != null){
+						remoteFiles.add(fileNameMap.get(correlationId));
+					}
+				}
+			}
+		});
+	}
+
+	/**
+	 * Attempts to upload a SdlFile to core
+	 * @param file SdlFile with file name and one of A) fileData, B) Uri, or C) resourceID set
+	 * @param listener called when core responds to the attempt to upload the file
+	 */
+	public void uploadFile(final SdlFile file, final CompletionListener listener){
+		PutFile putFile = createPutFile(file);
+
+		putFile.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if(response.getSuccess()){
+					remoteFiles.add(file.getName());
+				}
+				listener.onComplete(response.getSuccess());
+			}
+
+			@Override
+			public void onError(int correlationId, Result resultCode, String info) {
+				super.onError(correlationId, resultCode, info);
+				listener.onComplete(false);
+			}
+		});
+
+		internalInterface.sendRPCRequest(putFile);
+	}
+
+	/**
+	 * Attempts to upload a list of SdlFiles to core
+	 * @param files list of SdlFiles with file name and one of A) fileData, B) Uri, or C) resourceID set
+	 * @param listener callback that is called once core responds to all upload requests
+	 */
+	public void uploadFiles(List<? extends SdlFile> files, final MultipleFileCompletionListener listener){
+		if(files == null || files.isEmpty()){
+			return;
+		}
+		final List<PutFile> putFileRequests = new ArrayList<>();
+		for(SdlFile file : files){
+			putFileRequests.add(createPutFile(file));
+		}
+		sendMultipleFileOperations(putFileRequests, listener);
+	}
+
+	/**
+	 * Attempts to upload a SdlArtwork to core
+	 * @param file SdlArtwork with file name and one of A) fileData, B) Uri, or C) resourceID set
+	 * @param listener called when core responds to the attempt to upload the file
+	 */
+	public void uploadArtwork(final SdlArtwork file, final CompletionListener listener){
```

Comment:
I think that uploadArtwork() & uploadArtworks() are unnecessary and can be removed.  Because we can use uploadFile() & uploadFiles() to handle both SDLFile & SDLArtWork

Reply:
The [iOS library has these APIs](https://github.com/smartdevicelink/sdl_ios/blob/master/SmartDeviceLink/SDLFileManager.m#L424) so if we can stay consistent that is better.

Reply:
Ahh okay, it makes sense.